urts/src/lib.rs

```diff
@@ -2,22 +2,57 @@
 
 #![doc = include_str!("../README.md")]
 
-use mc_sgx_core_sys_types::sgx_status_t;
-use mc_sgx_urts_sys::{sgx_create_enclave_from_buffer_ex, sgx_destroy_enclave};
-use mc_sgx_urts_sys_types::sgx_enclave_id_t;
-use std::{ops::Deref, os::raw::c_int, ptr};
+use mc_sgx_core_types::{Error, TargetInfo};
+use mc_sgx_urts_sys::{
+    sgx_create_enclave_from_buffer_ex, sgx_destroy_enclave, sgx_get_target_info,
+    SGX_CREATE_ENCLAVE_EX_KSS, SGX_CREATE_ENCLAVE_EX_KSS_BIT_IDX, SGX_CREATE_ENCLAVE_EX_PCL,
+    SGX_CREATE_ENCLAVE_EX_PCL_BIT_IDX,
+};
+use mc_sgx_urts_sys_types::{sgx_enclave_id_t, sgx_kss_config_t};
+use mc_sgx_util::ResultInto;
+use std::{ffi::c_void, fs::File, io::Read, mem::MaybeUninit, os::raw::c_int, path::Path, ptr};
 
-#[derive(Debug, PartialEq, Eq)]
-pub enum Error {
-    // An error provided from the SGX SDK
-    SgxStatus(sgx_status_t),
+/// Structure defining configuration for Key Sharing and Separation
+pub struct KssConfig {
```

Comment:
I think we can keep `KssConfig` here, unless there's some use for it inside an enclave.

That said, this struct does need to be changed to wrap the bindgen type, because otherwise we have to keep them in sync.